frontend/src/components/TaskTableUnrated.tsx

```diff
@@ -91,21 +91,33 @@ const TableUnratedTaskRow: TableRow<Task> = ({ item: task, style }) => {
     }
   }, [task, allCustomers, allUsers, userInfo, type]);
 
-  const openModal = (
-    modalType: ModalTypes.RATE_TASK_MODAL | ModalTypes.NOTIFY_USERS_MODAL,
-  ) => (e: SyntheticEvent) => {
+  const openNotifyModal = (e: SyntheticEvent) => {
     e.preventDefault();
     e.stopPropagation();
     dispatch(
       modalsActions.showModal({
-        modalType,
+        modalType: ModalTypes.NOTIFY_USERS_MODAL,
         modalProps: {
           taskId: task.id,
         },
       }),
     );
   };
 
+  const openRateModal = (e: SyntheticEvent) => {
+    e.preventDefault();
+    e.stopPropagation();
+    dispatch(
+      modalsActions.showModal({
+        modalType: ModalTypes.RATE_TASK_MODAL,
+        modalProps: {
+          taskId: task.id,
+          edit: false,
+        },
+      }),
+    );
+  };
```

Comment:
```suggestion
  const openModal = (payload: ShowModalPayload) => (e: SyntheticEvent) => {
    e.preventDefault();
    e.stopPropagation();
    dispatch(modalsActions.showModal(payload));
  };

  const openRateModal = openModal({
    modalType: ModalTypes.RATE_TASK_MODAL,
    modalProps: {
      taskId: task.id,
      edit: false,
    },
  });
```

Could make the `openModal()` take `ShowModalPayload`. which would make it applicable for all three handlers.